arbor/util/piecewise.hpp

```diff
@@ -305,8 +319,16 @@ template <> struct pw_elements<void> {
         else return partn.index(x);
     }
 
-    using const_iterator = indexed_const_iterator<pw_elements<void>>;
-    using iterator = const_iterator;
+    // Return iterator pair spanning elements whose corresponding closed intervals contain x.
+    std::pair<iterator, iterator> equal_range(double x) const {
```

Comment:
This function could use a comment or two; it's not immediately obvious why shifting the iterators is needed and correct the way it is.